File: examples/python/core/logger/basic-usage.py
from aws_durable_execution_sdk_python import DurableContext, durable_execution

@durable_execution
def handler(event: dict, context: DurableContext) -> str:
    # Log at the top level
    context.logger.info("Starting workflow", extra={"event_id": event.get("id")})
    
    # Execute a step
    result: str = context.step(
        lambda _: "processed",
        name="process_data",
    )
    
    context.logger.info("Workflow completed", extra={"result": result})
    return result

File: examples/java/core/logger/basic-usage.java
public class BasicUsageHandler extends DurableHandler<Map<String, Object>, String> {

    @Override
    public String handleRequest(Map<String, Object> event, DurableContext ctx) {
        // Log at the top level
        ctx.getLogger().info("Starting workflow, event_id={}", event.get("id"));

        // Execute a step
        String result = ctx.step("process_data", String.class, stepCtx -> "processed");

        ctx.getLogger().info("Workflow completed, result={}", result);
        return result;
    }
}
